tests/lib/fde-setup-hook/fde-setup.go

```diff
@@ -130,23 +155,37 @@ func runFdeRevealKey() error {
 		return fmt.Errorf("fde-reveal-key input is not valid base64: %v", err)
 	}
 	if !bytes.Equal(decodedBase64Key, js.SealedKey) {
-		return fmt.Errorf("fde-reveal-key input is not strictly the same base64 decoded as binary decoded")
+		return fmt.Errorf("fde-reveal-key key input is not strictly the same base64 decoded as binary decoded")
+	}
+	decodedBase64Handle, err := base64.StdEncoding.DecodeString(jsStrict.Handle)
+	if err != nil {
+		return fmt.Errorf("fde-reveal-key handle input is not valid base64: %v", err)
+	}
+	if !bytes.Equal(decodedBase64Handle, js.Handle) {
+		return fmt.Errorf("fde-reveal-key handle input is not strictly the same base64 decoded as binary decoded")
 	}
 
 	switch js.Op {
 	case "reveal":
-		// strip the "USK$" prefix
-		unsealedKey := xor13(js.SealedKey[len("USK$"):])
-		fmt.Fprintf(os.Stdout, "%s", unsealedKey)
+		// check that the handle created in initial-setup is passed
+		// back to reveal correctly.
+		if string(js.Handle) != string(testKeyHandle) {
+			return fmt.Errorf(`fde-reveal-key expected handle %q but got %q`, testKeyHandle, js.Handle)
+		}
+		// "decrypt" key
+		var res fdeRevealKeyResultJSON
+		res.Key = xor13(js.SealedKey)
+		if err := json.NewEncoder(osStdout).Encode(res); err != nil {
+			return err
+		}
 	case "lock":
-		// nothing right now
-
-		// NOTE: when using this file as an example code for implementing a real
-		// world, production grade FDE hook, the lock operation must be
-		// implemented here to block decryption operations
+		// NOTE: when using this file as an example code for
+		// implementing a real world, production grade FDE
+		// hook, the lock operation must be implemented here
+		// to block decryption operations. This example does
+		// nothing.
 	case "features":
-		// XXX: Not used right now but might in the future?
-		fmt.Fprintf(os.Stdout, `{"features":[]}`)
+		fmt.Fprintf(osStdout, `{"features":[]}`)
 	default:
 		return fmt.Errorf(`unsupported operations %q`, js.Op)
```

Comment:
Is plural "operations" intended?

Reply:
Nice catch, I don't think this was intentional